gadopt/time_stepper.py

```diff
@@ -117,120 +124,209 @@ def advance(
         self.get_final_solution()
 
 
-class ERKGeneric(RungeKuttaTimeIntegrator):
-    """Generic explicit Runge-Kutta time integrator.
+class IrksomeIntegrator(TimeIntegratorBase):
+    """Time integrator using Irksome as the backend.
 
-    Implements the Butcher form. All terms in the equation are treated explicitly.
+    This class wraps Irksome's TimeStepper while maintaining G-ADOPT's API
+    for compatibility with our existing code.
 
-    Arguments:
-      equation: G-ADOPT equation to solve
-      solution: Firedrake function reperesenting the equation's solution
-      dt: Integration time step
-      solution_old: Firedrake function representing the equation's solution
+    Args:
+        equation: G-ADOPT equation to integrate
+        solution: Firedrake function representing the equation's solution
+        dt: Integration time step (Firedrake Constant or float)
+        butcher: Irksome Butcher tableau (e.g., GaussLegendre, RadauIIA)
+        stage_type: Type of stage formulation (e.g., "deriv")
+        solution_old: Firedrake function representing the equation's solution
                       at the previous timestep
-      solver_parameters: Dictionary of solver parameters provided to PETSc
-      strong_bcs: List of Firedrake Dirichlet boundary conditions
-
+        strong_bcs: List of Firedrake Dirichlet boundary conditions
+        bc_type: Boundary condition type for Irksome ("DAE" or "ODE")
+        solver_parameters: Dictionary of solver parameters provided to PETSc
     """
+
     def __init__(
         self,
         equation: Equation,
         solution: firedrake.Function,
         dt: float,
+        butcher: ButcherTableau,
+        stage_type: str = "deriv",
         solution_old: Optional[firedrake.Function] = None,
-        solver_parameters: Optional[dict[str, Any]] = {},
         strong_bcs: Optional[list[firedrake.DirichletBC]] = None,
+        bc_type: str = "DAE",
+        solver_parameters: Optional[dict[str, Any]] = None,
     ):
-        super(ERKGeneric, self).__init__(
-            equation, solution, dt, solution_old, solver_parameters, strong_bcs
+        super().__init__()
```

Comment:
This statement should not be needed because `TimeIntegratorBase` does not define `__init__`.